pkg/client/cache/watch_cache.go

```diff
@@ -27,12 +27,22 @@ import (
 	"k8s.io/kubernetes/pkg/watch"
 )
 
+// WatchCacheEvent is a single "watch event" that is send to users of
+// WatchCache. Additionally to a typical "watch.Event" it contains
+// the previous value of the object to enable proper filtering in the
+// upper layers.
```

Comment:
I'm afraid I don't understand what are you suggesting. The whole watch in apiserver is a very local change and in general people shouldn't worry about it all (I would consider it as part of storage, which generally developers don't care about while implementing higher-level features).